### PR TITLE
Add option to disable URL normalization

### LINKED ISSUE #123: Don't clean_url unless user desires
This is good small library, i integrated in my app to solve a problem mentioned [in this SO question](https://stackoverflow.com/questions/51641828/sendgrid-doesnt-display-deeplink-like-a-clickable-link), but only to find out that this library doesn't for my use case.

To keep it short, i want to shorten urls to app in form of something like: `intent://xyz` this is common pattern used for mobile apps, and for API based rails apps which are exposing APIs for their mobile apps.

A quick look suggests issue is here in this line: https://github.com/jpmcgrath/shortener/blob/develop/app/models/shortener/shortened_url.rb#L55
Here it is calling clean_url method, which is sort of pre-processing the URL before saving to DB.

What it ends up being is if i send: `myapp://some_path` to be shortened, it is saved as `/myapp://some_path` in DB which obviously doesn't redirect properly. And end up something like: `https://mydomain.com/myapp://some_path`
 which is incorrect.

Suggestion: Disable url_cleaning by default, and make it configurable through initialize. Or vice versa.

I am happy to submit a patch, let me know how you would like it to be. Thanks.
My point

### PR DESCRIPTION
**Pull request by ziaulrehman40**
Fixes #123

### CODE REVIEW

**Comment by ziaulrehman40** [comment]
@fschwahn Sounds good, but i guess we can do this in another iteration. For now this should be enough. Because cleaning URL is still possible on the app logic end. So we should be good.

File: spec/models/shortened_url_spec.rb [hunk 1]
```diff
@@ -177,8 +177,26 @@
         it 'finds the shortened url from slashless oath' do
           expect(Shortener::ShortenedUrl.generate!(path)).to eq existing_shortened_url
         end
-        it "should look up exsiting URL" do
-          expect(Shortener::ShortenedUrl.generate!("/#{path}")).to eq existing_shortened_url
+
+        context 'with auto_clean_url enabled by default' do
+          it "should look up existing cleaned URL" do
```
Comment (olleolleolle) [inline spec/models/shortened_url_spec.rb:182]:
```suggestion
          it "looks up existing cleaned URL" do
```

File: spec/models/shortened_url_spec.rb [hunk 2]
```diff
@@ -177,8 +177,26 @@
...
+
+        context 'with auto_clean_url disabled' do
+          around do |spec|
+            tries = Shortener.auto_clean_url
+            Shortener.auto_clean_url = false
+            spec.run
+            Shortener.auto_clean_url = tries
+          end
+
+          it "should not look up existing cleaned URL" do
```
Comment (olleolleolle) [inline spec/models/shortened_url_spec.rb:195]:
```suggestion
          it "does not look up existing cleaned URL" do
```

**Comment by jpmcgrath** [comment]
Released as v0.8.2: https://rubygems.org/gems/shortener/versions/0.8.2

**Pull request merged**